src/apps/bristleback_apps/bm_rbr/user_code/rbr_sensor.cpp

```diff
@@ -0,0 +1,234 @@
+#include "rbr_sensor.h"
+#include "FreeRTOS.h"
+#include "OrderedSeparatorLineParser.h"
+#include "bm_printf.h"
+#include "configuration.h"
+#include "payload_uart.h"
+#include "rbr_sensor_util.h"
+#include "serial.h"
+#include "stm32_rtc.h"
+#include "task_priorities.h"
+#include "uptime.h"
+#include <ctype.h>
+#include <math.h>
+#include <string.h>
+
+extern cfg::Configuration *systemConfigurationPartition;
+
+/*!
+* @brief Initialize the RBR sensor driver.
+* @param type The sensor type.
+*/
+void RbrSensor::init(BmRbrDataMsg::SensorType_t type) {
+  _type = type;
+  _stored_type = type;
+  for (int i = 0; i < NUM_PARSERS; i++) {
+    if (_parsers[i]) {
+      _parsers[i]->init();
+    }
+  }
+  PLUART::init(USER_TASK_PRIORITY);
+  // Baud set per expected baud rate of the sensor.
+  PLUART::setBaud(BAUD_RATE);
+  // Disable passing raw bytes to user app.
+  PLUART::setUseByteStreamBuffer(false);
+  // Enable parsing lines and passing to user app.
+  /// Warning: PLUART only stores a single line at a time. If your attached payload sends lines
+  /// faster than the app reads them, they will be overwritten and data will be lost.
+  PLUART::setUseLineBuffer(true);
+  // Set a line termination character per protocol of the sensor.
+  PLUART::setTerminationCharacter(LINE_TERM);
+  // Turn on the UART.
+  PLUART::enable();
+}
+
+/*!
+* @brief Probe the sensor type.
+* @param timeout_ms The timeout in milliseconds.
+* @return True if the sensor type was successfully probed, false otherwise.
+*/
+bool RbrSensor::probeType(uint32_t timeout_ms) {
```

Comment:
Polling of the RBR sensor to get its type. If there is no config in flash, or a new RBR sensor is swapped in - save it to config. 
If we can't read the sensor type 3 times in a row - set the type to unknown.